qiskit/circuit/quantumcircuit.py

```diff
@@ -272,6 +277,19 @@ def __init__(
         self.unit = "dt"
         self.metadata = {} if metadata is None else metadata
 
+    def _new_data(self, iterable: Iterable[CircuitInstruction] | None = None):
```

Comment:
super minor, but `()` is a safe immutable default to use for a value of type `Iterable[CircuitInstruction]` that avoids needing a special case